packages/backend/src/corporate-proxy.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 The Janus IDP Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import { bootstrap } from 'global-agent';
+import { Agent, ProxyAgent, Dispatcher, setGlobalDispatcher } from 'undici';
+
+/**
+ * Adds support for corporate proxy to both 'node-fetch' (using 'global-agent') and native 'fetch' (using 'undici') packages.
+ *
+ * Ref: https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/help-im-behind-a-corporate-proxy.md
+ * Ref: https://gist.github.com/zicklag/1bb50db6c5138de347c224fda14286da (to support 'no_proxy')
+ */
+export function configureCorporateProxyAgent() {
+  // Bootstrap global-agent, which addresses node-fetch proxy-ing.
+  // global-agent purposely uses namespaced env vars to prevent conflicting behavior with other libraries,
+  // but user can set GLOBAL_AGENT_ENVIRONMENT_VARIABLE_NAMESPACE to an empty value for global-agent to use
+  // the conventional HTTP_PROXY, HTTPS_PROXY and NO_PROXY environment variables.
+  // More details in https://github.com/gajus/global-agent#what-is-the-reason-global-agentbootstrap-does-not-use-http_proxy
+  bootstrap();
+
+  // Configure the undici package, which affects the native 'fetch'. It leverages the same env vars used by global-agent,
+  // or the more conventional HTTP(S)_PROXY ones.
+  const proxyEnv =
```

Comment:
Yes, they are processed based on the protocol, hostname and port. The check here is only to determine whether to create the global dispatcher, as long as any of these env vars is set. So the order of precedence shouldn't matter, I think.
However, what can be confusing is the order of precedence between, say, `HTTP_PROXY` and `http_proxy` if both are set. I couldn't find a clear answer to this, as it turns out that it depends on how each library handles that. I documented setting the upper case form because that's what both 'global-agent` honors (and Axios as well).